bothan-coingecko/src/worker/asset_worker.rs

```diff
@@ -0,0 +1,142 @@
+use std::sync::Weak;
+use std::time::Duration;
+
+use chrono::NaiveDateTime;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::Decimal;
+use tokio::time::{interval, timeout, Interval};
+use tracing::{info, warn};
+
+use bothan_core::store::Store;
+use bothan_core::types::AssetInfo;
+
+use crate::api::rest::CoinGeckoRestAPI;
+use crate::api::types::{Market, Order};
+use crate::worker::error::ParseError;
+use crate::worker::CoinGeckoWorker;
+
+pub(crate) fn start_asset_worker(
+    weak_worker: Weak<CoinGeckoWorker>,
+    mut update_interval: Interval,
+    page_size: usize,
+    page_query_delay: Duration,
+) {
+    tokio::spawn(async move {
+        while let Some(worker) = weak_worker.upgrade() {
+            info!("updating asset info");
```

Comment:
will these logs prefixed with coingecko somewhere? do we need to creating tracing span? same for binance. can be a separate PR